[jit] fix python enumerate with start kwarg

yuguo68 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #80585

fix https://github.com/pytorch/pytorch/issues/80150
turns out we have a unittest for this case but there is a typo so the test does not run. 

With this fix both enumerate(x, start=1) and enumerate(x, 1) are supported.

yuguo68 committed:
[jit] fix python enumerate with start kwarg

[ghstack-poisoned]

davidberard98 commented on torch/csrc/jit/frontend/ir_emitter.cpp:
```diff
@@ -3577,11 +3579,21 @@ struct to_ir {
+              << "enumerate expected at most 2 arguments, got " << arg_size;
         }
+
+        if (attribute_size == 1) {
+          if (attributes[0].name().name() != "start") {
+            throw ErrorReport(loc)
+                << "enumerate expected kwarg name 'start', got '"
+                << attributes[0].name().name() << "'";
+          }
+          start_index = emitSugaredExpr(attributes[0].value(), 1)->asValue(loc, method);
+        }
```
so is it true that `attributes` just means `kwargs`?

yuguo68 replied:
yes, I think so. in the same file, we have https://github.com/pytorch/pytorch/blob/c71886e04827f9fa87a3a4a0ee4e76f3b1dc22e1/torch/csrc/jit/frontend/ir_emitter.cpp#L3350

davidberard98 commented on test/test_jit.py:
```diff
@@ -11765,15 +11765,25 @@ def fn(x):
             return sum
 
-        self.checkScript(fn, ([1, 2, 3, 4, 5],))
+        self.checkScript(fn_enumerate_start_kwarg, ([1, 2, 3, 4, 5],))
```
whoops... thanks for catching this!!

davidberard98 reviewed (approved):
Awesome, thanks! Looks good to me

yuguo68 committed:
lint on "[jit] fix python enumerate with start kwarg"


fix https://github.com/pytorch/pytorch/issues/80150
turns out we have a unittest for this case but there is a typo so the test does not run. 

With this fix both enumerate(x, start=1) and enumerate(x, 1) are supported. 

[ghstack-poisoned]

yuguo68 commented:
@pytorchbot merge
